### ISSUE TITLE
[enet] Config files, usage, building as shared library

### ISSUE THREAD

**Issue by Spirrwell** [microsoft/vcpkg#20230]
TL;DR: I would be open to doing a pull request to export enetConfig cmake files, add some usage to go along with that, and support BUILD_SHARED_LIBS so it may be built as a shared library.

---

For reference, this is somewhat related to this issue here: https://github.com/microsoft/vcpkg/issues/19317 But I'm referring to CMake exported config files in this post, not pkg-config.

Currently, the enet port does not export a config, nor does it support pkg-config. (I'm not familiar with the latter) It also doesn't emit usage information which isn't ideal.

The port also doesn't appear to properly support building enet as a shared library, even though this is supported by the library, just not in their CMakeLists.txt file.

The issue linked mentions moving the specified ports to their native buildsystems, however enet has multiple ways to be built, even through CMake. The current port simply overwrites the CMakeLists.txt file with its own, and the other buildsystems would likely need to be "fixed" or patched as well in order to work with vcpkg anyway.

I think ideally these problems would be dealt with on enet's repository. I don't know if they are open to that or not.

**Comment by JonLiu1993** [comment]
@Spirrwell ,We welcome you to submit a  pr, if you can discuss it with enet Upstream in advance, it would be better

**Comment by Spirrwell** [comment]
I will go ahead and open an issue on their repository and see if they're open to changes to their CMakeLists.txt to be more vcpkg friendly. However their last release was in 2020 and I don't see any significant incoming changes.

So I'll probably go ahead and do a pull request anyway here to tackle these issues so that it can still be tied to their last release if that's alright. 

Then later, if they do a new release with a more friendly CMakeLists, that could be used instead of overwriting it from the vcpkg port.

**Comment by PhoebeHui** [comment]
@Spirrwell, thanks for your contribution!
